File: samples/jvm/src/main/java/dev/pubgkt/StatsSample.java
package dev.pubgkt;

import java.util.List;

import kotlin.coroutines.EmptyCoroutineContext;
import kotlinx.coroutines.BuildersKt;

import dev.pubgkt.players.GetPlayersByNameKt;
import dev.pubgkt.players.Player;
import dev.pubgkt.stats.lifetime.GetLifetimeStatsKt;
import dev.pubgkt.stats.lifetime.LifetimeGameModeStats;
import dev.pubgkt.stats.lifetime.LifetimePlayerStats;
import dev.pubgkt.stats.ranked.GetRankedStatsKt;
import dev.pubgkt.stats.ranked.RankedStats;
import dev.pubgkt.stats.season.GetSeasonStatsKt;
import dev.pubgkt.stats.season.SeasonGameModeStats;
import dev.pubgkt.stats.season.SeasonPlayerStats;

public class StatsSample {
    public static void main(String[] args) throws InterruptedException {
        if (args.length == 0) {
            throw new IllegalArgumentException("Usage: StatsSample <api-key>");
        }

        PubgApi api = new PubgApi(args[0]);

        List<Player> players = BuildersKt.runBlocking(
                EmptyCoroutineContext.INSTANCE,
                (_, cont) -> GetPlayersByNameKt.getPlayersByNames(api, List.of("sparkingg", "TGLTN"), Platform.STEAM, cont)
        );
        String accountId = players.get(0).getId();
        List<String> accountIds = players.stream().map(Player::getId).toList();

        List<Season> seasons = BuildersKt.runBlocking(
                EmptyCoroutineContext.INSTANCE,
                (_, cont) -> SeasonKt.seasons(api, Platform.STEAM, cont)
        );
        String seasonId = seasons.stream().filter(Season::isCurrentSeason).findFirst().orElseThrow().getId();

        System.out.println("=== getLifetimeStatsByAccountId ===");
        LifetimePlayerStats lifetime = BuildersKt.runBlocking(
                EmptyCoroutineContext.INSTANCE,
                (_, cont) -> GetLifetimeStatsKt.getLifetimeStatsByAccountId(api, accountId, Platform.STEAM, cont)
        );
        System.out.println("Player: " + lifetime.getPlayerId());
        System.out.println("Squad FPP — Wins: " + lifetime.getSquadFpp().getWins() + ", Kills: " + lifetime.getSquadFpp().getKills());

        System.out.println("\n=== getLifetimeStatsByGameModeAndPlayers ===");
        List<LifetimeGameModeStats> lifetimeByMode = BuildersKt.runBlocking(
                EmptyCoroutineContext.INSTANCE,
                (_, cont) -> GetLifetimeStatsKt.getLifetimeStatsByGameModeAndPlayers(api, GameMode.SQUAD_FPP, accountIds, Platform.STEAM, cont)
        );
        lifetimeByMode.forEach(s -> System.out.println(s.getPlayerId() + ": " + s.getStats().getKills() + " kills"));

        System.out.println("\n=== getSeasonStatsByAccountId ===");
        SeasonPlayerStats seasonStats = BuildersKt.runBlocking(
                EmptyCoroutineContext.INSTANCE,
                (_, cont) -> GetSeasonStatsKt.getSeasonStatsByAccountId(api, accountId, seasonId, Platform.STEAM, cont)
        );
        System.out.println("Season: " + seasonStats.getSeasonId());
        System.out.println("Squad FPP — Wins: " + seasonStats.getSquadFpp().getWins() + ", Kills: " + seasonStats.getSquadFpp().getKills());

        System.out.println("\n=== getSeasonStatsByGameModeAndPlayers ===");
        List<SeasonGameModeStats> seasonByMode = BuildersKt.runBlocking(
                EmptyCoroutineContext.INSTANCE,
                (_, cont) -> GetSeasonStatsKt.getSeasonStatsByGameModeAndPlayers(api, seasonId, GameMode.SQUAD_FPP, accountIds, Platform.STEAM, cont)
        );
        seasonByMode.forEach(s -> System.out.println(s.getPlayerId() + ": " + s.getStats().getKills() + " kills"));

        System.out.println("\n=== getRankedStatsByAccountIdAndSeasonId ===");
        RankedStats ranked = BuildersKt.runBlocking(
                EmptyCoroutineContext.INSTANCE,
                (_, cont) -> GetRankedStatsKt.getRankedStatsByAccountIdAndSeasonId(api, accountId, seasonId, Platform.STEAM, cont)
        );
        System.out.println("Player: " + ranked.getPlayerId());
        System.out.println("Tier: " + ranked.getCurrentTier() + ", RP: " + ranked.getCurrentRankPoint() + ", KDA: " + ranked.getKda());
    }
}

File: samples/jvm/src/main/kotlin/dev/pubgkt/StatsSample.kt
package dev.pubgkt

import dev.pubgkt.players.getPlayersByNames
import dev.pubgkt.stats.lifetime.getLifetimeStatsByAccountId
import dev.pubgkt.stats.lifetime.getLifetimeStatsByGameModeAndPlayers
import dev.pubgkt.stats.ranked.getRankedStatsByAccountIdAndSeasonId
import dev.pubgkt.stats.season.getSeasonStatsByAccountId
import dev.pubgkt.stats.season.getSeasonStatsByGameModeAndPlayers

suspend fun main(args: Array<String>) {
    val apiKey = args.getOrNull(0) ?: error("Usage: StatsSample <api-key>")
    val api = PubgApi(apiKey)

    val players = api.getPlayersByNames("sparkingg", "TGLTN")
    val accountId = players.first().id
    val accountIds = players.map { it.id }

    val seasons = api.seasons(Platform.STEAM)
    val currentSeason = seasons.currentOrNull() ?: error("No current season found")

    println("=== getLifetimeStatsByAccountId ===")
    val lifetime = api.getLifetimeStatsByAccountId(accountId)
    println("Player: ${lifetime.playerId}")
    println("Squad FPP — Wins: ${lifetime.squadFpp.wins}, Kills: ${lifetime.squadFpp.kills}")

    println("\n=== getLifetimeStatsByGameModeAndPlayers ===")
    val lifetimeByMode = api.getLifetimeStatsByGameModeAndPlayers(
        gameMode = GameMode.SQUAD_FPP,
        accountIds = accountIds,
    )
    lifetimeByMode.forEach { println("${it.playerId}: ${it.stats.kills} kills, ${it.stats.wins} wins") }

    println("\n=== getSeasonStatsByAccountId ===")
    val seasonStats = api.getSeasonStatsByAccountId(accountId, seasonId = currentSeason.id)
    println("Season: ${seasonStats.seasonId}")
    println("Squad FPP — Wins: ${seasonStats.squadFpp.wins}, Kills: ${seasonStats.squadFpp.kills}")

    println("\n=== getSeasonStatsByGameModeAndPlayers ===")
    val seasonByMode = api.getSeasonStatsByGameModeAndPlayers(
        seasonId = currentSeason.id,
        gameMode = GameMode.SQUAD_FPP,
        accountIds = accountIds,
    )
    seasonByMode.forEach { println("${it.playerId}: ${it.stats.kills} kills, ${it.stats.wins} wins") }

    println("\n=== getRankedStatsByAccountIdAndSeasonId ===")
    val ranked = api.getRankedStatsByAccountIdAndSeasonId(accountId, seasonId = currentSeason.id)
    println("Player: ${ranked.playerId}")
    println("Tier: ${ranked.currentTier}, RP: ${ranked.currentRankPoint}, KDA: ${ranked.kda}")
}
